### DISCUSSION TITLE
Input text is lost/disappears while typing fast and switching to another input.

### DISCUSSION THREAD

**Discussion by auhale0** [livewire/livewire#4709] [1. Help]
Hi,

I have tried everything possible for a smiple form with livewire to work without losing input text while switching between inputs, but it still loses data.

I have attached a video below so that you see what I mean exactly.
FYI: The video I shared is using wired:model on all inputs.

[Livewire forms problems while typing fast between inputs](https://www.filemail.com/d/vuvjtlsxghmepiq)

Things I have tried:
1- Adding wired:model.lazy
2- Adding wired:modle.defer
3- Creating a separate project from scratch and have a form with livewire following this tutorial: [Create form submission example using laravel with livewire](https://webomnizz.com/laravel-livewire-form-submission-example/#install-livewire-package.)
4- Following livewire troubleshoot suggestions:
[Livewire troubleshooting - DOM diff issues](https://laravel-livewire.com/docs/2.x/troubleshooting)

Yet, there's still input text loss/disappearance while typing fast and switching between inputs.

My environtment:
Browser: Chrome / Edge
OS: windows 10
PHP 7.4
Laravel 8
Liverwire 2.2.5

**Comment by auhale0** [comment]
Update:
It seems to be working if I update all inputs  (not just one or two) to wired:model.defer, but the question remains
why doesn't wired:model.lazy or wired:model work on a simple form?

The problem with defer,  I'll lose the real-time validation.

Thanks

**Comment by joshhanley** [comment]
@auhale0 can you add your video directly to GitHub (upload it in a comment) so we don't have to download it to view it? Thanks!

**Comment by auhale0** [comment]
@joshhanley Thanks for pointing it out I didn't see the drag & drop.


Here you go this is the video:
As you can see if you type and switch you will see that I lost the word Jackson in the first name. All I'm using is normal contact form with normal wired:model. 
Another thing to notice is that the last letter or few letters they disappear and come back in some cases and sometimes disappear completely too.

https://user-images.githubusercontent.com/99243560/155068005-b9f9e1fb-d38a-4e73-91f7-044c9a3a9599.mp4

**Comment by silnarm** [comment]
Asynchronous requests clobbering each other would be my guess. Unless Livewire is queueing responses, which I have not noticed (though I haven't perused the JS much yet).
The `dbounce` modifier is probably the one you want, and should fix it (if the multitude of requests is indeed the issue):
```
<input wire:model.debounce.350ms="foo" />
```

**Comment by joshhanley** [comment]
@auhale0 yeah @silnarm is correct. You should put `.defer` on your inputs or if you need real time validation, then add a debounce as suggested.

Hope this helps!

**Comment by OtiZ2** [comment]
I had the same problem.
In my case it was because I had:

`public string $firstname;`


And in mount I only filled it when it was defined:

```php
public function mount()
{
    if ($user->firstname) {
        $this->firstname = $user->firstname;
    }
}
```

My issue was solved when I started always filling it.


```php
public ?string $firstname;
public function mount()
{
    $this->firstname = $user->firstname ?? null;
}
```
